duckduckgo/iOS
# Fix bug where VPN location isn't respected

Task/Issue URL: https://app.asana.com/0/0/1206025077480506/f
BSK: https://github.com/duckduckgo/BrowserServicesKit/pull/576

**Description**:
There was a bug with location switching where the user’s choice wasn’t always being respected. I noticed that there was a place where we weren’t passing it. I also noticed that we’re sometimes not using the selected environment because of the default function arguments, so I also updated them as it’s a somewhat related issue.

**Steps to test this PR**:
1. Just make sure NetP / Geoswitching still works. Perhaps try it with slow network to push it a bit 

**Copy Testing**:


**Orientation Testing**:


**Device Testing**:


**OS Testing**:


**Theme Testing**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
